tests/e2e/framework/framework.go

```diff
@@ -0,0 +1,192 @@
+package framework
+
+import (
+	"flag"
+	"github.com/golang/glog"
+	"io/ioutil"
+	"os"
+	"sync"
+)
+
+var (
+	logsBucketPath = flag.String("logs_bucket_path", "", "Cloud Storage Bucket path to use to store logs")
+)
+
+const (
+	TMP_PREFIX = "istio.e2e."
+)
+
+type TestInfo struct {
+	RunId         string
+	TestId        string
+	LogBucketPath string
+	LogsPath      string
+}
+
+type TestCleanup struct {
+	Cleanables         []Cleanable
+	CleanablesLock     sync.Mutex
+	CleanupActions     []func() error
+	CleanupActionsLock sync.Mutex
+}
+
+type CommonConfig struct {
+	Cleanup *TestCleanup
+	Info    *TestInfo
+	// Other common config here.
+}
+
+type Cleanable interface {
+	SetUp() error
+	TearDown() error
+}
+
+type Runnable interface {
+	Run() int
+}
+
+func NewCommonConfig(testId string) *CommonConfig {
+	tmpDir, err := ioutil.TempDir(os.TempDir(), TMP_PREFIX)
+	if err != nil {
+		glog.Fatal("Could not create a Temporary dir")
```

Comment:
Fatal failures in a library function is an anti-pattern. Better to return an error and let the caller deal with it.